src/cluster/multi_step_cluster.cpp

```diff
@@ -53,46 +56,116 @@ vector<int> MultiStep::cluster(DatabaseFile &db, const vector<bool> *filter) {
 	opt.db = &db;
 	opt.self = true;
 	Neighbors nb(db.ref_header.sequences);
+	nb.index.resize(nb.size());
+	for (size_t i = 0; i< nb.index.size(); i++)
+		nb.index[i] = i;
+
 	opt.consumer = &nb;
 	opt.db_filter = filter;
 
 	Workflow::Search::run(opt);
 
 	message_stream << "Edges = " << nb.edges.size() << endl;
 
+	for (size_t k = 0; k < nb.size(); k++) {
+		vector<uint32_t> seen_nodes;
```

Comment:
Diesen Vektor solltest du außerhalb der Schleife deklarieren, weil er sonst in jedem Schleifendurchlauf neu initialisiert wird.